src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,99 @@
+     *   composition: >
+     *     Round-Trip Modals are completed by a small well defined sequence of 2 steps max. that might be displayed on a sequence of different modals connected through some “next” Button.
+     *   effect: >
+     *     Round-Trip Modals perform sub-workflow involving some kind of user input. Sub-workflow is completed and user is returned to starting point allowing for continuing the overriding workflow.
+     *
+     * rules:
+     *   usage:
+     *     1: >
+     *       Round-Trip Modals MUST contain at least two buttons at the bottom of the modals: a button to cancel (right) the workflow and a button to finish or reach the next step in the workflow (left).
+     *     2: >
+     *       Round-Trip Modals SHOULD be used, if the user would lose the context otherwise. If the action can be performed within the same context (e.g. add a post in a forum, edit a wiki page), a Round-Trip Modal MUST NOT be used.
+     *     3: >
+     *       When the workflow is completed, Round-Trip Modals SHOULD show the same view as was displayed when initiating the Modal.
+     *     4: >
+     *       Round-Trip Modals SHOULD NOT be used to add new items of any kind since adding item is a linear workflow redirecting to the newly added item setting- or content-tab.
+     *     5: >
+     *       Round-Trip Modals SHOULD NOT be used to perform complex workflows.
+     *
+     * ---
+     * @param string $title
+     * @param Component\Component $content
+     * @return \ILIAS\UI\Component\Modal\RoundTrip
+     */
+    public function roundtrip($title, Component\Component $content);
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The Lightbox Modal displays a full of data such as documents, images or a movie inside a Lightbox.
+     *   composition: >
+     *     If multiple pages or images are to be displayed in one Lightbox Modal, they can flipped through.
+     *   effect: >
+     *      Lightbox Modals are activated by clicking the Full View Glyphicon, the Title of the object or it’s Thumbnail.
+     *
+     * rules:
+     *   usage:
+     *     1: >
+     *       Lightbox Modals MUST contain a Closing Button on the top right and a title above the presented item.
+     *     2: >
+     *       Lightobx Modals MUST contain a title above the presented item.
+     *     3: >
+     *       Lightbox Modals SHOULD contain a descriptional text below the presented items.
+     *     4: >
+     *       Multiple images or pages inside a Lightbox Modals MUST be presented in carousel like manner allowing to flickr through items.
+     *
+     * ---
+     * @param string $title
+     * @param Component\Component $content
+     * @return \ILIAS\UI\Component\Modal\Lightbox
+     */
+    public function lightbox($title, Component\Component $content);
```

Comment:
@wanze So far we decided against it. However, at anytime when you pass content of indifferent type this might be helfpful (e.g. a panel with some collection of items used as content). The reason we did not do it yet, is that we hope to be more specific than just a list. Note that there is a function toArray in component helper, allowing you to cast quickly to array, no matter if an array is passed or not. In panels this is used like

`
	/**
	 * @param string $title
	 * @param \ILIAS\UI\Component\Component[] | \ILIAS\UI\Component\Component $content
	 */
	public function __construct($title,$content) {
		$this->checkStringArg("title",$title);
		$content = $this->toArray($content);
		$types = [C\Component::class];
		$this->checkArgListElements("content", $content, $types);

		$this->title = $title;
		$this->content = $content;
	}
`